steps/redhat.py

```diff
@@ -10,6 +10,16 @@
 rh_gpg_fingerprint = '567E 347A D004 4ADE 55BA  8A5F 199E 2F91 FD43 1D51'
 rh_gpg_fingerprint_short = '199E2F91FD431D51'
 
+def get_ipaddr(context, host):
+    r = context.remote_cmd("setup",
+                            host,
+                            module_args="filter=ansible_default_ipv4")
+    if r:
+        ip_addr = r[0]['ansible_facts']['ansible_default_ipv4']['address']
+        return ip_addr
+    else:
+        assert False, "Couldn't obtain host IP address"
+
```

Comment:
The problem is that if r is not defined, its value is False (as opposed to the successful call where we have embedded dict inside a list) so return statement would end with a traceback.